tools/Customization/DevHome.FileExplorerSourceControlIntegration/SourceControlProvider.cs

```diff
@@ -66,6 +69,20 @@ internal ILocalRepositoryProvider GetLocalProvider(string rootPath)
         Log.Information("GetLocalProvider succeeded");
         return provider;
     }
+
+    IDictionary<string, object> IExtraFolderPropertiesHandler.GetProperties(string[] propertyStrings, string rootFolderPath, string relativePath)
+    {
+        var localProvider = GetLocalProvider(rootFolderPath);
+        var localProviderResult = localProvider.GetRepository(rootFolderPath);
+        if (localProviderResult.Result.Status == ProviderOperationStatus.Failure)
+        {
+            log.Information("Could not open local repository.");
+            log.Information(localProviderResult.Result.DisplayMessage);
+            throw new ArgumentOutOfRangeException(nameof(rootFolderPath), rootFolderPath, localProviderResult.Result.DisplayMessage);

```

Comment:
Why ArgumentOutOfRange?  Isn't rootFolderPath a string?

Reply:
rootFolderPath is, indeed a string. I may have overthought this exception type. `ArgumentException` seems more appropriate.